src/main/host/syscall/type_formatting.rs

```diff
@@ -341,3 +403,32 @@ impl<const LEN_INDEX: usize> SyscallDisplay for SyscallVal<'_, SyscallSockAddrAr
         write!(f, "{addr}")
     }
 }
+
+impl SyscallDisplay for SyscallVal<'_, *const libc::msghdr> {
+    fn fmt(
+        &self,
+        f: &mut std::fmt::Formatter<'_>,
+        options: FmtOptions,
+        mem: &MemoryManager,
+    ) -> std::fmt::Result {
+        let ptr = self.reg.into();
+
+        if options == FmtOptions::Deterministic {
+            return write!(f, "<pointer>");
+        }
+
+        // read the msghdr
+        let ptr = TypedArrayForeignPtr::new::<libc::msghdr>(ptr, 1);
+        let Ok(msg) = mem.memory_ref(ptr) else {
+            // if we couldn't read the memory, just show the pointer instead
+            return write!(f, "{:p}", ptr.ptr());
```

Comment:
Maybe something like `"bad:{:p}"` to make it clearer that the pointer couldn't be read?

Reply:
I'll do this in a follow-up PR, since there's a bunch of places this would need to be updated to be consistent.